sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -361,6 +362,41 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
   }
 
   /**
+   * Loads an `Dataset[String]` storing CSV rows and returns the result as a `DataFrame`.
+   *
+   * Unless the schema is specified using `schema` function, this function goes through the
+   * input once to determine the input schema.
+   *
+   * @param csvDataset input Dataset with one CSV row per record
+   * @since 2.2.0
+   */
+  def csv(csvDataset: Dataset[String]): DataFrame = {
```

Comment:
shall we also add `def json(lines: Dataset[String])`? and deprecate `json(r: RDD[String])` and `json(r: JavaRDD[String])`

cc @rxin